meteor/server/api/snapshot.ts

```diff
@@ -681,74 +688,127 @@ export async function restoreFromRundownPlaylistSnapshot(
 	})
 
 	// List any ids that need updating on other documents
-	const rundownIdMap: { [key: string]: RundownId } = {}
+	const rundownIdMap = new Map<RundownId, RundownId>()
 	const getNewRundownId = (oldRundownId: RundownId) => {
-		return rundownIdMap[unprotectString(oldRundownId)]
+		const rundownId = rundownIdMap.get(oldRundownId)
+		if (!rundownId) {
+			throw new Meteor.Error(500, `Could not find new rundownId for "${oldRundownId}"`)
+		}
+		return rundownId
 	}
 	_.each(snapshot.rundowns, (rd) => {
```

Comment:
Since you're touching the code anyway I guess I'd like to see _.each replaced by for..of